File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int>mapy;
        int n=nums.size();
        for(int i=0;i<n;i++){
            int rem=target-nums[i];
            if(mapy.find(rem)!=mapy.end()){
                return {i,mapy[rem]};
            }else{
                mapy[nums[i]]=i;
            }
        }
        return{};
    }
};

File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n=nums.length;
        HashMap<Integer,Integer>mapy = new HashMap<>();
        for(int i=0;i<n;i++){
            int a = target - nums[i];//a = target - b
            if(mapy.containsKey(a)){
                return new int[]{mapy.get(a),i};
            }
            mapy.put(nums[i],i);
        }
        return new int[]{-1,-1};
    }
}